### PR TITLE
Issues/3392 fix signedout reader

### LINKED ISSUE #3392: Reader: Show comments and likes for signed-out users
Currently all action buttons are hidden when using the signed-out reader.  We need to : 
- Show, but disable, the like meta button
- Show the comments button and display the comments list when tapped.  
- Comment cells should also display like counts, but not the like button, or the reply button. 
- Comment screen should not display the reply field.

Additionally, if a post has comments closed do not show the reply field on the comment screen, and only show the comment button if there are comments to view.

### PR DESCRIPTION
**Pull request by aerych**
Closes #3392 

When a user is signed out the reader will now
- Show the like button and count but the button is disabled.
- Show the comments button and display the comments list when tapped.
- Comment cells display like counts, but not the like button, or the reply button.
- Comment screen does not display the reply field.

Additionally, for a signed in reader where a post has comments closed 
- the reply field on the comment screen is not shown, 
- the comment action button is only shown with the post if there are existing comments.

### CODE REVIEW

**Comment by aerych** [comment]
cc @jleandroperez Would you mind having a look at this one since comments are shared with Notifications?

File: WordPress/Classes/ViewRelated/Reader/CommentContentView.m [hunk 1]
```diff
@@ -118,22 +127,71 @@ - (void)configureConstraints
...
+                                                                  constant:0.0]];
+
+    self.commentMetaHeightConstraint = [NSLayoutConstraint constraintWithItem:self.commentMeta
+                                                                    attribute:NSLayoutAttributeHeight
+                                                                    relatedBy:NSLayoutRelationEqual
+                                                                       toItem:nil
+                                                                    attribute:nil
+                                                                   multiplier:1.0
+                                                                     constant:CommentContentViewMetaHeight];
+    [self addConstraint:self.commentMetaHeightConstraint];
```
Comment (jleandroperez) [inline WordPress/Classes/ViewRelated/Reader/CommentContentView.m]: Interface Builder, someday! :D

File: WordPress/Classes/ViewRelated/Reader/CommentContentView.m [hunk 2]
```diff
@@ -219,6 +280,15 @@ - (WPRichTextView *)viewForContent
     return textContentView;
 }
 
+- (UIView *)viewForCommentMeta
+{
+    UIView *view = [[UIView alloc] initWithFrame:CGRectMake(0.0, 0.0, 100.0, CommentContentViewMetaHeight)];
```
Comment (jleandroperez) [inline WordPress/Classes/ViewRelated/Reader/CommentContentView.m]: Annoying comment: maybe something like this?
```
static CGSize const CommentContentViewMetaSize = { 100.0f, 16.0f };
```

File: WordPress/Classes/ViewRelated/Reader/ReaderCommentsViewController.m [hunk 3]
```diff
@@ -1094,7 +1107,7 @@ - (void)handleReplyTapped:(id<WPContentViewProvider>)contentProvider
         return;
     }
 
-    if (![self canComment]) {
+    if (!self.canComment) {
```
Comment (jleandroperez) [inline WordPress/Classes/ViewRelated/Reader/ReaderCommentsViewController.m]: Properties FTW!

File: WordPress/Classes/ViewRelated/Reader/ReaderCommentsViewController.m [hunk 4]
```diff
@@ -526,6 +531,16 @@ - (void)setAvatarForComment:(Comment *)comment forCell:(ReaderCommentCell *)cell
...
 
+- (void)checkIfCanComment
+{
+    self.canComment = self.post.commentsOpen && self.isLoggedIn;
+}
+
+- (void)checkIfLoggedIn
+{
+    NSManagedObjectContext *context = [[ContextManager sharedInstance] mainContext];
+    self.isLoggedIn = [[[AccountService alloc] initWithManagedObjectContext:context] defaultWordPressComAccount] != nil;
```
Comment (jleandroperez) [inline WordPress/Classes/ViewRelated/Reader/ReaderCommentsViewController.m]: (Definitely no need to change anything!). I'm starting to think that we could use a helper, with the default dotcom user always accessible + perhaps all of the accounts. We got this snippet in few other places!

**Comment by jleandroperez** [comment]
@aerych UX detail to think about. In the Post Details view, the **Comments** pushes the Comments List.

However, the Like button does nothing. I'm wondering if that might be confusing (why can't i like a post?). Maybe we should set the button's alpha so it's clear that it's disabled?. (Or display an alert saying.. please sign into dotcom?).

Other than that, :shipit: !!

File: WordPress/Classes/ViewRelated/Reader/CommentContentView.m (hunk 1 above)
Comment (aerych) [inline WordPress/Classes/ViewRelated/Reader/CommentContentView.m]: OMG! Constraints in code are so ugly now. IB FTW!

File: WordPress/Classes/ViewRelated/Reader/CommentContentView.m (hunk 2 above)
Comment (aerych) [inline WordPress/Classes/ViewRelated/Reader/CommentContentView.m]: Hmm... Well, in this case the frame is arbitrary and really disposable (constraints are gonna do what constraints are gonna do!). I included the height constant cos I thought it added some context.  I'll add a comment about the frame being arbitrary.

File: WordPress/Classes/ViewRelated/Reader/ReaderCommentsViewController.m (hunk 4 above)
Comment (aerych) [inline WordPress/Classes/ViewRelated/Reader/ReaderCommentsViewController.m]: Agreed! I'll open an issue to create a "isLoggedIn" helper (on WPAccount maybe?) and we can scrub all the things.

**Comment by aerych** [comment]
Thanks @jleandroperez!  I'm going to open a separate issue for the like button UX. Good thought about that!

**Comment by jleandroperez** [comment]
@aerych pure awesomeness, thanks Eric!

**Pull request merged**